WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt

```diff
@@ -200,10 +207,19 @@ class ReaderDiscoverLogic @Inject constructor(
     @Suppress("NestedBlockDepth")
     private fun createSimplifiedJson(cardsJsonArray: JSONArray, discoverTasks: DiscoverTasks): JSONArray {
         val simplifiedJsonList = mutableListOf<JSONObject>()
-        var firstYouMayLikeCard: JSONObject? = null
+        var firstRecommendationCard: JSONObject? = null
         for (i in 0 until cardsJsonArray.length()) {
             val cardJson = cardsJsonArray.getJSONObject(i)
-            when (cardJson.getString(JSON_CARD_TYPE)) {
+            // We should not have a recommended blogs or interests/tags card as the first element on Discover feed.
+            val cardType = cardJson.optString(JSON_CARD_TYPE, "")
+            val isCardTypeRecommendation =
+                cardType == JSON_CARD_RECOMMENDED_BLOGS || cardType == JSON_CARD_INTERESTS_YOU_MAY_LIKE
+            val isFirstPage = discoverTasks == REQUEST_FIRST_PAGE
```

Comment:
Is it possible to move this check outside of the `for` loop?

Reply:
Good catch! Done.